src/Domain.LinnApps/PurchaseOrders/PurchaseOrderService.cs

```diff
@@ -0,0 +1,29 @@
+﻿namespace Linn.Purchasing.Domain.LinnApps.PartSuppliers
+{
+    using System.Collections.Generic;
+
+    using Linn.Common.Authorisation;
+    using Linn.Purchasing.Domain.LinnApps.PurchaseOrders;
+
+    public class PurchaseOrderService : IPurchaseOrderService
+    {
+        private readonly IAuthorisationService authService;
+
+        public PurchaseOrderService(
+            IAuthorisationService authService)
+        {
+            this.authService = authService;
+        }
+
+        public void UpdatePurchaseOrder(PurchaseOrder current, PurchaseOrder updated, IEnumerable<string> privileges)
+        {
+            if (!this.authService.HasPermissionFor(AuthorisedAction.PartSupplierUpdate, privileges))
+            {
+                throw new UnauthorisedActionException("You are not authorised to Overbook records");
```

Comment:
Would probably reword to 'You are not authorised to allow overbooks on a PO' or something